lib/visitor-keys.js

```diff
@@ -1,3 +1,10 @@
+/**
+ * @typedef {{ readonly [type: string]: ReadonlyArray<string> }} KeysLooseReadonly
+ */
```

Comment:
Why is this "Loose"? This definition looks the same as the definition for `KeysStrictReadonly` in the other file.

Reply:
Yes, you're right. This was a relic I had been kind of leaving in place in case there were concerns about our dropping the `| undefined` portion so we could go back to the Loose version at least here if desired (the discrepancy mentioned in the PR description) (`unionWith` couldn't use it with the code as currently factored). I've added a commit to update this (and if we don't need to permit `undefined` as `@types/eslint-visitor-keys` does, then I could also rename these types to stop calling them "Strict" too since this was only added to be in contrast to the "Loose" one, now having been dropped).

Reply:
Are there any observable differences in typescript checks when `| undefined` is added, or is it just redundant? I'd expect that it is implied that map-like objects cannot have all keys and that therefore `obj[key]` can be undefined.

Reply:
It gives an error with an explicit `undefined` (which is avoided when added to the type):

https://www.typescriptlang.org/play?#code/DYUwLgBAhgXBDeBYAUBNEBOIoBMD2AdsAJ4QDaYxADiHAM5gYCWBA5gLpwBK2+RxAQQwYoxADwNmbAHwoAvhAC8CFOmhwArgRwgAZixA55AbhRA

I

Reply:
Ok, sure. I just didn't know if that had been allowed for the sake of downstream libraries being able to compose objects with these properties removed or something. But I suppose that even if there would be some case for this, they could presumably manage this per their own project's needs. I can go ahead and change then.